addons/website_sale/controllers/main.py

```diff
@@ -934,11 +934,14 @@ def tracking_cart(self, **post):
         return ret
 
     @http.route(['/shop/get_unit_price'], type='json', auth="public", methods=['POST'], website=True)
-    def get_unit_price(self, product_ids, add_qty, **kw):
+    def get_unit_price(self, product_ids, add_qty, on_cart=False, **kw):
         cr, uid, context, pool = request.cr, request.uid, request.context, request.registry
         products = pool['product.product'].browse(cr, uid, product_ids, context=context)
         partner = pool['res.users'].browse(cr, uid, uid, context=context).partner_id
-        pricelist_id = request.session.get('sale_order_code_pricelist_id') or partner.property_product_pricelist.id
+        if on_cart:
+            pricelist_id = request.session.get('sale_order_code_pricelist_id') or partner.property_product_pricelist.id
+        else:
+            pricelist_id = partner.property_product_pricelist.id
```

Comment:
pricelist_id = use_order_pricelist and request.session.get('sale_order_code_pricelist_id') or partner.property_product_pricelist.id